// @mantine/emotion/cjs/index.cjs
'use strict';

var react = require('@emotion/react');
var createStyles = require('./create-styles.cjs');
var emotionTransform = require('./emotion-transform.cjs');
var Global = require('./Global.cjs');
var MantineEmotionProvider = require('./MantineEmotionProvider.cjs');
var useCss = require('./use-css.cjs');
var ssr = require('./ssr.cjs');



Object.defineProperty(exports, "keyframes", {
	enumerable: true,
	get: function () { return react.keyframes; }
});
exports.createStyles = createStyles.createStyles;
exports.getHelpers = createStyles.getHelpers;
exports.getStylesRef = createStyles.getStylesRef;
exports.emotionTransform = emotionTransform.emotionTransform;
exports.Global = Global.Global;
exports.EmotionCacheContext = MantineEmotionProvider.EmotionCacheContext;
exports.MantineEmotionProvider = MantineEmotionProvider.MantineEmotionProvider;
exports.useEmotionCache = MantineEmotionProvider.useEmotionCache;
exports.cssFactory = useCss.cssFactory;
exports.useCss = useCss.useCss;
exports.ServerStyles = ssr.ServerStyles;
exports.createGetInitialProps = ssr.createGetInitialProps;
exports.getSSRStyles = ssr.getSSRStyles;
